### PR TITLE
Support background effects in config screen

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
1. New effects button in config screen
2. Side pane opens on config screen

Screenshot 2023-03-27 at 10 42 13 PM

Screenshot 2023-03-27 at 10 42 28 PM

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3193136
https://skype.visualstudio.com/SPOOL/_workitems/edit/3193138

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/styles/CallConfiguration.styles.ts [hunk 1]
```diff
@@ -125,3 +127,25 @@ export const startCallButtonStyleMobile = mergeStyles({
   width: '100%',
   maxWidth: 'unset'
 });
+
+/* @conditional-compile-remove(video-background-effects) */
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/styles/CallConfiguration.styles.ts:131]: We don't have to CC it if that is not exported publicly, and it is encouraged to reduce CC comments if possible

**Review by PorterNan — APPROVED** [review]
There is a concern that we should keep main branch always working without any blank panel/not working UI part

So I would expect check this in after we have the side effect pane content merged in first

Approved, and look forward to your PR to merge side effect pane content

**Comment by prabhjot-msft** [comment]
> There is a concern that we should keep main branch always working without any blank panel/not working UI part
> 
> So I would expect check this in after we have the side effect pane content merged in first
> 
> Approved, and look forward to your PR to merge side effect pane content

Thanks for bringing that up, I have added the content to the pane, here is the PR https://github.com/Azure/communication-ui-library/pull/2858/

File: packages/react-composites/src/composites/CallComposite/styles/CallConfiguration.styles.ts (hunk 1 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/styles/CallConfiguration.styles.ts]: done

File: packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx [hunk 2]
```diff
@@ -270,6 +291,22 @@ export const ConfigurationPage = (props: ConfigurationPageProps): JSX.Element =>
                 {title}
                 {callDescription}
               </Stack.Item>
+              {
+                /* @conditional-compile-remove(video-background-effects) */
+                <DefaultButton
+                  iconProps={{ iconName: 'OptionsVideoBackgroundEffect', styles: { root: { lineHeight: 0 } } }}
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx:297]: store the styles in a styles file. This component will re-render needlessly if objects are not static or memoized.

Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx]: Looks like that styling is not needed, icon is rendering well without that as well, removed it.

File: packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx [hunk 3]
```diff
@@ -270,6 +291,22 @@ export const ConfigurationPage = (props: ConfigurationPageProps): JSX.Element =>
...
+                /* @conditional-compile-remove(video-background-effects) */
+                <DefaultButton
+                  iconProps={{ iconName: 'OptionsVideoBackgroundEffect' }}
+                  styles={effectsButtonStyles(theme)}
+                  onClick={() => {
+                    if (showVideoEffectsPane) {
+                      setShowVideoEffectsPane(false);
+                    } else {
+                      setShowVideoEffectsPane(true);
+                    }
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx:304]: Maybe can simplify this to just:
```
setVideoEffectsPane(!showVideoEffectsPane)
```

File: packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx [hunk 4]
```diff
@@ -75,6 +85,15 @@ export const ConfigurationPage = (props: ConfigurationPageProps): JSX.Element =>
     /* @conditional-compile-remove(call-readiness) */ onNetworkingTroubleShootingClick
   } = props;
 
+  /* @conditional-compile-remove(video-background-effects) */
+  const [showVideoEffectsPane, setVideoEffectsPane] = useState(false);
+  /* @conditional-compile-remove(video-background-effects) */
+  const setShowVideoEffectsPane = useCallback(
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx:91]: nit: I don't think you need this callback. You already have the useState setter.

**Review by mgamis-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx (hunk 3 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx]: done

File: packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx (hunk 4 above)
Comment (prabhjot-msft) [reply packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx]: done

**Pull request merged**